quiz_app_main.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+require_relative 'teacher'
+require_relative 'student'
+# class for controlling all flow
+class Main
+  def initialize
+    @storage = { 'Teacher' => [], 'Student' => [] }
```

Comment:
Let's keep it array.
```suggestion
    @storage = []
```
We can simply check user instance by using `is_a?` method.
i.e, `@storage.first.is_a? Teacher` or `@storage.first.is_a? Student`